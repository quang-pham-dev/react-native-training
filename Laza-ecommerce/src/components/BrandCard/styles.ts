import { StyleSheet } from 'react-native';

// Themes
import Colors from 'themes/Colors';
import Metrics from 'themes/Metrics';

const styles = StyleSheet.create({
  brandCardContainer: {
    justifyContent: 'center',
  },
  brandCardWrapper: {
    flexDirection: 'row',
    alignItems: 'center',
    borderRadius: 10,
    height: 50,
    backgroundColor: Colors.secondaryBackground,
    marginRight: Metrics.margin.sm,
    paddingRight: Metrics.padding.sm,
  },
  brandLogoWrapper: {
    width: 40,
    height: 40,
    padding: Metrics.padding.xs,
    borderRadius: 10,
    marginLeft: Metrics.margin.xsm,
    marginRight: Metrics.margin.sm,
    marginVertical: Metrics.margin.xsm,
    backgroundColor: Colors.primaryBackground,
  },
  brandLogo: {
    width: '100%',
    height: '100%',
    resizeMode: 'contain',
  },
});

export default styles;
