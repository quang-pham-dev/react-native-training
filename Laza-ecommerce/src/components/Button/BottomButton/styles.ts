import { StyleSheet } from 'react-native';
// Styles
import { Metrics, Colors, Fonts } from '@styles/themes/index';

const styles = StyleSheet.create({
  button: {
    paddingVertical: Metrics.padding.md,
    paddingHorizontal: Metrics.padding.xl,
    alignItems: 'center',
  },
  title: {
    color: Colors.white,
    textTransform: 'none',
    fontFamily: Fonts.fontFamily.Inter_600SemiBold,
    fontSize: Fonts.size.sm,
    lineHeight: Fonts.lineHeight.lg,
  },
});

export default styles;
