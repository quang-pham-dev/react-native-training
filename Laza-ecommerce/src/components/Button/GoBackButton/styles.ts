// Styles
import { Metrics } from '@styles/themes/index';
import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  button: {
    marginTop: Metrics.margin.customXxl,
    marginLeft: Metrics.margin.lg,
    alignItems: 'center',
  },
  icon: {
    width: Metrics.icons.semiLarge,
    height: Metrics.icons.semiLarge,
  },
});

export default styles;
