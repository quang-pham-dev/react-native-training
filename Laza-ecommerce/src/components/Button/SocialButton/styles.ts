import { StyleSheet } from 'react-native';
// Styles
import { Metrics, Colors, Fonts } from '@styles/themes/index';

const styles = StyleSheet.create({
  button: {
    paddingVertical: Metrics.padding.md,
    paddingHorizontal: Metrics.padding.xl,
    alignItems: 'center',
    borderColor: Colors.transparent,
    borderWidth: 1,
    backgroundColor: Colors.white,
  },
  title: {
    color: Colors.white,
    textTransform: 'none',
    fontFamily: Fonts.fontFamily.Inter_600SemiBold,
    fontSize: Fonts.size.sm,
    lineHeight: Metrics.height.custom,
  },
  icon: {
    width: Metrics.icons.custom,
    height: Metrics.icons.custom,
  },
});

export default styles;
