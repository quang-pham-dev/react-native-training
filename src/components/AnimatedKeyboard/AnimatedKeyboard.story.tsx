import React from 'react'
import {TextInput} from 'react-native'
import {storiesOf} from '@storybook/react-native'

import {StoryScreen, Story, UseCase} from '../../../storybook/views'

import AnimatedKeyboard from './index'

declare let module: NodeModule

storiesOf('AnimatedKeyboard', module)
  .addDecorator(fn => <StoryScreen>{fn()}</StoryScreen>)
  .add('Default', () => (
    <Story>
      <UseCase text="default">
        <AnimatedKeyboard>
          <TextInput placeholder="Enter your Username" />
        </AnimatedKeyboard>
      </UseCase>
    </Story>
  ))
