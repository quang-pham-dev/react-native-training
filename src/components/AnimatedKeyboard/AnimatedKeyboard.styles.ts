import styled from 'styled-components/native'

export const AnimatedKeyboardWrapperStyled = styled.ScrollView.attrs({
  contentContainerStyle: {
    flex: 1,
    'justify-content': 'center',
    'align-items': 'center',
  },
})``
