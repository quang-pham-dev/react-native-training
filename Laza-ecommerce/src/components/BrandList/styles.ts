import { StyleSheet } from 'react-native';
import { Metrics } from 'styles/themes';
const styles = StyleSheet.create({
  container: {},
  brandTitle: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    paddingBottom: Metrics.padding.xm,
  },
});

export default styles;
