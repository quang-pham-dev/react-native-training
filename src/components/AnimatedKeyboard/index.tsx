import React, {PropsWithChildren} from 'react'

import Animated, {
  useAnimatedKeyboard,
  useAnimatedStyle,
} from 'react-native-reanimated'

import {AnimatedKeyboardWrapperStyled} from './AnimatedKeyboard.styles'

const AnimatedKeyboard = ({children}: PropsWithChildren) => {
  const keyboard = useAnimatedKeyboard()
  const animatedStyle = useAnimatedStyle(() => {
    return {
      transform: [{translateY: -keyboard.height.value}],
    }
  })

  return (
    <AnimatedKeyboardWrapperStyled>
      <Animated.View style={[animatedStyle]}>{children}</Animated.View>
    </AnimatedKeyboardWrapperStyled>
  )
}

export default AnimatedKeyboard
