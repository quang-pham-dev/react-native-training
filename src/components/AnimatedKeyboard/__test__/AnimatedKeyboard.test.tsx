import React from 'react'
import {render} from '@testing-library/react-native'

import {AnimatedKeyboardWrapperStyled} from '../AnimatedKeyboard.styles'
import {TextInput} from 'react-native'

describe('AnimatedKeyboard', () => {
  test('Should render correctly', () => {
    const component = render(
      <AnimatedKeyboardWrapperStyled>
        <TextInput placeholder="Enter your Username" />
      </AnimatedKeyboardWrapperStyled>,
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
