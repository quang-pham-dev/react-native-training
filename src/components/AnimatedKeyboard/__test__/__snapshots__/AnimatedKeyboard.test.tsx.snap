// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimatedKeyboard Should render correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "align-items": "center",
      "flex": 1,
      "justify-content": "center",
    }
  }
  style={
    Array [
      Object {},
    ]
  }
>
  <View>
    <TextInput
      placeholder="Enter your Username"
    />
  </View>
</RCTScrollView>
`;
